src/dash-table/syntax-tree/lexeme/relational.ts

```diff
@@ -73,6 +83,24 @@ export const greaterThan: IUnboundedLexeme = R.merge({
     regexp: /^(>|gt)/i
 }, LEXEME_BASE);
 
+const DATE_OPTIONS: IDateValidation = {
+    allow_YY: true
+};
+
+export const likeDate: IUnboundedLexeme = R.merge({
+    evaluate: relationalEvaluator(([op, exp]) => {
+        const normalizedOp = normalizeDate(op, DATE_OPTIONS);
+        const normalizedExp = normalizeDate(exp, DATE_OPTIONS);
+
+        return !R.isNil(normalizedOp) &&
+            !R.isNil(normalizedExp) &&
+            // IE11 does not support `startsWith`
+            normalizedOp.indexOf(normalizedExp) === 0;
+    }),
+    subType: RelationalOperator.LikeDate,
+    regexp: /^(like)/i
```

Comment:
still not sure about naming this just `like` - at some point we may want a more sql-like `like` supporting wildcard string matching - or even if we don't add that and instead add a more powerful regex matcher, users may expect `like` to have that kind of behavior. Anyway this one is so tightly coupled to dates, it really feels like it needs to reference date in its name.

`datelike` / `likedate` / `datehas`? @chriddyp thoughts? For context, this is what @Marc-Andre-Rivet and I ended up with as the default operator for dates: match as many fields as you provide in the filter, with anything afterward - ie `like '2011-05'` matches any day, any time, in May 2011, while `like '2011-05-03 12'` matches anything in the 12:00 hour of May 3, 2011.

Reply:
Hm..
- `datecontains`? I suppose some folks might think "Wednesday" would be valid here (capturing all Wednesdays across all time)
    - Although, perhaps the "Wednesday" filter could be "dateperiod"?
- `datewithin`?

Reply:
> datecontains? I suppose some folks might think "Wednesday" would be valid here (capturing all Wednesdays across all time)

Variations on this discussed with Alex previously. Later on it should be possible to add functions or transformations that would allow stuff like that, e.g.

`dayOfWeek({dateField}) eq 3` or `month({dateField}) eq 10` -- going with numbers just not to get into a i18n/l10n/filtering discussion

Reply:
Right, we'll create lots of other functions later to do more complex and precise filtering. This one will be used almost exclusively as the default operator, so it doesn't need to have a particularly efficient name since hardly anyone will type it. But it does need to *have* a name, as it would be strange if there were behavior you can get with a simple filter that you can't replicate at all with the more complex syntax.

OK, so one more try: `datestartswith`? Still lets us add `startswith` later for strings, but makes clear that it behaves differently for dates, and as I said I don't think it needs to be short, better to be explicit.